src/library/filescanner.c

```diff
@@ -539,13 +605,58 @@ create_virtual_path(char *path, char *virtual_path, int virtual_path_len)
   return 0;
 }
 
+/*
+ * Returns informations about the attributes of the file at the given 'path' in the structure
+ * pointed to by 'sb'.
+ *
+ * If path is a symbolic link, the attributes in sb describe the file that the link points to
+ * and resolved_path contains the resolved path (resolved_path must be of length PATH_MAX).
+ * If path is not a symbolic link, resolved_path holds the same value as path.
+ *
+ * The return value is 0 if the operation is successful, or -1 on failure. In addition
+ */
+static int
+read_attributes(const char *path, struct stat *sb, char *resolved_path)
```

Comment:
I have no real opinion on this matter, just stumbled across the guidelines from google for c/c++ code which define it the other way around: https://google.github.io/styleguide/cppguide.html#Function_Parameter_Ordering